lib/urn.rb

```diff
@@ -1,5 +1,6 @@
 class URN
-  PATTERN = 'urn:(?!urn:)[a-z0-9\-]{1,31}:[\S]+'.freeze
+  PATTERN = %(urn:(?!urn:)[A-Za-z0-9][A-Za-z0-9\-]{1,31}:) +
```

Comment:
Brilliant! we even fixed another issue I just noticed, we were allowing upper-case `urn` string in the NID too (e.g. `urn:URN:123`)